tokio-threadpool/src/worker/entry.rs

```diff
@@ -201,7 +202,14 @@ impl WorkerEntry {
     /// them into `dest` in order to balance the work distribution among
     /// workers.
     pub fn steal_tasks(&self, dest: &Self) -> deque::Steal<Arc<Task>> {
-        self.stealer.steal_many(&dest.worker)
+        use deque::Steal::*;
+        match self.stealer.steal_many(&dest.worker) {
+            Data(task) => Data(task),
+            Empty | Retry => match self.inbound.try_pop() {
+                None => Empty,
```

Comment:
Should this propagate `Retry` if that's what `steal_many()` returned?

Reply:
Yes, good catch.